`TSError: ⨯ Unable to compile TypeScript:` on latest commit

With current commit, when `make docker-run` it shows an error as below:

```
% make docker-run
docker run node-gitlab-2-github:latest

> gitlab-2-github@0.1.5 start
> node node_modules/ts-node/dist/bin.js ./src/index.ts

/app/node_modules/ts-node/src/index.ts:750
    return new TSError(diagnosticText, diagnosticCodes);
           ^
TSError: ⨯ Unable to compile TypeScript:
src/index.ts(55,36): error TS2345: Argument of type 'typeof throttling' is not assignable to parameter of type 'OctokitPlugin'.
  Types of parameters 'octokit' and 'octokit' are incompatible.
    Type 'import("/app/node_modules/@octokit/rest/node_modules/@octokit/core/dist-types/index").Octokit' is not assignable to type 'import("/app/node_modules/@octokit/core/dist-types/index").Octokit'.
      The types of 'request.defaults(...).defaults(...).defaults(...).endpoint.DEFAULTS' are incompatible between these types.
        Type 'object & O & O & O & import("/app/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestParameters").RequestParameters & { baseUrl: string; method: import("/app/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestMethod").RequestMethod; url?: string; headers: import("/app/node...' is not assignable to type 'object & O & O & O & import("/app/node_modules/@octokit/types/dist-types/RequestParameters").RequestParameters & { baseUrl: string; method: import("/app/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestMethod").RequestMethod; url?: string; headers: import("/app/node_modules/@octokit/types/dis...'.
          Type 'object & O & O & O & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }' is not assignable to type '{ baseUrl: string; method: RequestMethod; url?: string; headers: RequestHeaders & { accept: string; "user-agent": string; }; mediaType: { format: string; previews: string[]; }; }'.
            Types of property 'mediaType' are incompatible.
              Type '{ format?: string; previews?: string[]; } & { format?: string; previews?: string[]; } & { format: string; previews?: string[]; }' is not assignable to type '{ format: string; previews: string[]; }'.
                Property 'previews' is optional in type '{ format?: string; previews?: string[]; } & { format?: string; previews?: string[]; } & { format: string; previews?: string[]; }' but required in type '{ format: string; previews: string[]; }'.

    at createTSError (/app/node_modules/ts-node/src/index.ts:750:12)
    at reportTSError (/app/node_modules/ts-node/src/index.ts:754:19)
    at getOutput (/app/node_modules/ts-node/src/index.ts:941:36)
    at Object.compile (/app/node_modules/ts-node/src/index.ts:1243:30)
    at Module.m._compile (/app/node_modules/ts-node/src/index.ts:1370:30)
    at Module._extensions..js (node:internal/modules/cjs/loader:1252:10)
    at Object.require.extensions.<computed> [as .ts] (/app/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (node:internal/modules/cjs/loader:1076:32)
    at Function.Module._load (node:internal/modules/cjs/loader:911:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12) {
  diagnosticText: "src/index.ts(55,36): error TS2345: Argument of type 'typeof throttling' is not assignable to parameter of type 'OctokitPlugin'.\n" +
    "  Types of parameters 'octokit' and 'octokit' are incompatible.\n" +
    `    Type 'import("/app/node_modules/@octokit/rest/node_modules/@octokit/core/dist-types/index").Octokit' is not assignable to type 'import("/app/node_modules/@octokit/core/dist-types/index").Octokit'.\n` +
    "      The types of 'request.defaults(...).defaults(...).defaults(...).endpoint.DEFAULTS' are incompatible between these types.\n" +
    `        Type 'object & O & O & O & import("/app/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestParameters").RequestParameters & { baseUrl: string; method: import("/app/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestMethod").RequestMethod; url?: string; headers: import("/app/node...' is not assignable to type 'object & O & O & O & import("/app/node_modules/@octokit/types/dist-types/RequestParameters").RequestParameters & { baseUrl: string; method: import("/app/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestMethod").RequestMethod; url?: string; headers: import("/app/node_modules/@octokit/types/dis...'.\n` +
    `          Type 'object & O & O & O & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }' is not assignable to type '{ baseUrl: string; method: RequestMethod; url?: string; headers: RequestHeaders & { accept: string; "user-agent": string; }; mediaType: { format: string; previews: string[]; }; }'.\n` +
    "            Types of property 'mediaType' are incompatible.\n" +
    "              Type '{ format?: string; previews?: string[]; } & { format?: string; previews?: string[]; } & { format: string; previews?: string[]; }' is not assignable to type '{ format: string; previews: string[]; }'.\n" +
    "                Property 'previews' is optional in type '{ format?: string; previews?: string[]; } & { format?: string; previews?: string[]; } & { format: string; previews?: string[]; }' but required in type '{ format: string; previews: string[]; }'.\n",
  diagnosticCodes: [ 2345 ]
}
make: *** [docker-run] Error 1
```

I think it has something with this commit.
126dd9f4de009c1fd0352a6dcbd543ecec6ecffe

When I tried to revert to below, it ran successfully.
84a62dbcb9cc00e300dccab5dd87c1ad5a492f1d

```
% git reset --hard 84a62dbcb9cc00e300dccab5dd87c1ad5a492f1d
HEAD is now at 84a62db Merge pull request #219 from mschoettle/add-weburl

% make build-image
% make docker-run
```

Environment:
macOS 15.3.1
Docker version 24.0.2, build cb74dfc

Please check it:)

Not using docker. Same issue. 

npm version 10.9.0
node version 22.11.0

Just in case it helps narrow anything. 

```
/home/userName/node-gitlab-2-github/node_modules/ts-node/src/index.ts:750
    return new TSError(diagnosticText, diagnosticCodes);
           ^
TSError: ⨯ Unable to compile TypeScript:
src/index.ts:55:36 - error TS2345: Argument of type 'typeof throttling' is not assignable to parameter of type 'OctokitPlugin'.
  Types of parameters 'octokit' and 'octokit' are incompatible.
    Type 'import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/rest/node_modules/@octokit/core/dist-types/index").Octokit' is not assignable to type 'import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/core/dist-types/index").Octokit'.
      The types of 'request.defaults(...).defaults(...).defaults(...).endpoint.DEFAULTS' are incompatible between these types.
        Type 'object & O & O & O & import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestParameters").RequestParameters & { ...; }' is not assignable to type 'object & O & O & O & import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/types/dist-types/RequestParameters").RequestParameters & { baseUrl: string; method: import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestMethod").RequestMetho...'.
          Type 'object & O & O & O & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }' is not assignable to type '{ baseUrl: string; method: RequestMethod; url?: string; headers: RequestHeaders & { accept: string; "user-agent": string; }; mediaType: { format: string; previews: string[]; }; }'.
            Types of property 'mediaType' are incompatible.
              Type '{ format?: string; previews?: string[]; } & { format?: string; previews?: string[]; } & { format: string; previews?: string[]; }' is not assignable to type '{ format: string; previews: string[]; }'.
                Property 'previews' is optional in type '{ format?: string; previews?: string[]; } & { format?: string; previews?: string[]; } & { format: string; previews?: string[]; }' but required in type '{ format: string; previews: string[]; }'.

55 const MyOctokit = GitHubApi.plugin(throttling);
                                      ~~~~~~~~~~

    at createTSError (/home/goofballtech/node-gitlab-2-github/node_modules/ts-node/src/index.ts:750:12)
    at reportTSError (/home/goofballtech/node-gitlab-2-github/node_modules/ts-node/src/index.ts:754:19)
    at getOutput (/home/goofballtech/node-gitlab-2-github/node_modules/ts-node/src/index.ts:941:36)
    at Object.compile (/home/goofballtech/node-gitlab-2-github/node_modules/ts-node/src/index.ts:1243:30)
    at Module.m._compile (/home/goofballtech/node-gitlab-2-github/node_modules/ts-node/src/index.ts:1370:30)
    at node:internal/modules/cjs/loader:1689:10
    at Object.require.extensions.<computed> [as .ts] (/home/goofballtech/node-gitlab-2-github/node_modules/ts-node/src/index.ts:1374:12)
    at Module.load (node:internal/modules/cjs/loader:1318:32)
    at Function._load (node:internal/modules/cjs/loader:1128:12)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14) {
  diagnosticText: "\x1B[96msrc/index.ts\x1B[0m:\x1B[93m55\x1B[0m:\x1B[93m36\x1B[0m - \x1B[91merror\x1B[0m\x1B[90m TS2345: \x1B[0mArgument of type 'typeof throttling' is not assignable to parameter of type 'OctokitPlugin'.\n" +
    "  Types of parameters 'octokit' and 'octokit' are incompatible.\n" +
    `    Type 'import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/rest/node_modules/@octokit/core/dist-types/index").Octokit' is not assignable to type 'import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/core/dist-types/index").Octokit'.\n` +
    "      The types of 'request.defaults(...).defaults(...).defaults(...).endpoint.DEFAULTS' are incompatible between these types.\n" +
    `        Type 'object & O & O & O & import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestParameters").RequestParameters & { ...; }' is not assignable to type 'object & O & O & O & import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/types/dist-types/RequestParameters").RequestParameters & { baseUrl: string; method: import("/home/goofballtech/node-gitlab-2-github/node_modules/@octokit/rest/node_modules/@octokit/types/dist-types/RequestMethod").RequestMetho...'.\n` +
    `          Type 'object & O & O & O & RequestParameters & { baseUrl: string; method: RequestMethod; url?: string; headers: RequestHeaders & { ...; }; mediaType: { ...; }; }' is not assignable to type '{ baseUrl: string; method: RequestMethod; url?: string; headers: RequestHeaders & { accept: string; "user-agent": string; }; mediaType: { format: string; previews: string[]; }; }'.\n` +
    "            Types of property 'mediaType' are incompatible.\n" +
    "              Type '{ format?: string; previews?: string[]; } & { format?: string; previews?: string[]; } & { format: string; previews?: string[]; }' is not assignable to type '{ format: string; previews: string[]; }'.\n" +
    "                Property 'previews' is optional in type '{ format?: string; previews?: string[]; } & { format?: string; previews?: string[]; } & { format: string; previews?: string[]; }' but required in type '{ format: string; previews: string[]; }'.\n" +
    '\n' +
    '\x1B[7m55\x1B[0m const MyOctokit = GitHubApi.plugin(throttling);\n' +
    '\x1B[7m  \x1B[0m \x1B[91m                                   ~~~~~~~~~~\x1B[0m\n',
  diagnosticCodes: [ 2345 ]
}
```

--- a/Dockerfile
+++ b/Dockerfile
@@ -6,7 +6,7 @@
 #
 # If you're interested in testing other base images, take a look at this reference:
 # https://github.com/BretFisher/nodejs-rocks-in-docker
-FROM node:16-bullseye-slim
+FROM node:22-bullseye-slim
 
 ARG USERNAME=migrator
 ARG USER_UID=2000

--- a/package.json
+++ b/package.json
@@ -27,9 +27,9 @@
   "homepage": "https://github.com/piceaTech/node-gitlab-2-github#readme",
   "dependencies": {
     "@gitbeaker/node": "^35.1.0",
-    "@octokit/plugin-throttling": "^3.5.2",
+    "@octokit/plugin-throttling": "^9.4.0",
     "@octokit/rest": "^21.1.1",
-    "@octokit/types": "^6.34.0",
+    "@octokit/types": "^13.8.0",
     "aws-sdk": "^2.1053.0",
     "axios": "^1.6.0",
     "mime-types": "^2.1.34",

--- a/src/index.ts
+++ b/src/index.ts
@@ -68,21 +68,21 @@ const githubApi = new MyOctokit({
   },
   auth: 'token ' + settings.github.token,
   throttle: {
-    onRateLimit: async (retryAfter, options) => {
+    onRateLimit: async (retryAfter, options, octokit, retryCount) => {
       console.log(
         `Request quota exhausted for request ${options.method} ${options.url}`
       );
       console.log(`Retrying after ${retryAfter} seconds!`);
       return true;
     },
-    onAbuseLimit: async (retryAfter, options) => {
+    onSecondaryRateLimit: async (retryAfter, options, octokit, retryCount) => {
       console.log(
-        `Abuse detected for request ${options.method} ${options.url}`
-      );
+        `SecondaryRateLimit detected for request ${options.method} ${options.url}`
+      )
       console.log(`Retrying after ${retryAfter} seconds!`);
       return true;
     },
-    minimumAbuseRetryAfter: 1000,
+    fallbackSecondaryRateRetryAfter: 600,
   },
 });
 

